features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapSinkConsumer.java

```diff
@@ -120,7 +120,7 @@ private Log toLog(TrapLogDTO messageLog) {
 						eachMessage,
 						messageLog.getSystemId(),
 						messageLog.getLocation(),
-						messageLog.getTrapAddress());
+						messageLog.getTrapAddress(),config.shouldUseAddressFromVarbind());
```

Comment:
It's possible that user might have different configurations on Minion and OpenNMS ( in case they are only using Minion to receive traps, they might ignore updating trapd-configuration.xml) so instead of using `config.shouldUseAddressFromVarbind()` we should transmit the config option through TrapLogDTO and use that option here.